blocks/algorithms/__init__.py

```diff
@@ -293,23 +297,26 @@ class CompositeRule(StepRule):
     def __init__(self, components):
         self.components = components
 
-    def compute_steps(self, gradients):
-        result = gradients
+    def compute_steps(self, previous_steps):
+        steps = previous_steps
         updates = []
         for rule in self.components:
-            result, more_updates = rule.compute_steps(result)
+            steps, more_updates = rule.compute_steps(steps)
             updates += more_updates
-        return result, updates
+        return steps, updates
+
 
+class Scale(StepRule):
+    """A step in the direction proportional to the previous step.
 
-class SteepestDescent(StepRule):
-    """A step in the direction proportional to the gradient.
+    If the previous step is the gradient itself, this step rule is
+    equivalent to steepest ascent.
```

Comment:
Okay, this a question of how the documentation of step rules should be written: describing what they actually do or what the effect is when applied in the `GradientDescent`. Ideally both aspects should be attended. I think that in this case the first line of your docstring describes clearly what exactly the rule does. Then, when I see this

> If the previous step is the gradient itself, this step rule is equivalent to steepest ascent. 

, I expect it to be the description of what this step rule effectively does when used in the `GradientDescent`. This is why I was confused when I saw _ascent_.

I suggest to reformulate it like this:

> If used in `GradientDescent` alone, this step rule effectively does steepest descent.